airflow/models/baseoperator.py

```diff
@@ -672,10 +672,20 @@ def _set_xcomargs_dependencies(self) -> None:
 
         """
         from airflow.models.xcom_arg import XComArg
-        for field in self.template_fields:
-            arg = getattr(self, field)
+
+        def apply_set_upstream(arg: Any):
             if isinstance(arg, XComArg):
```

Comment:
this traversing part should be very "error prune". wee are starting to traverse all user templated args with no exception (old code is affected), it can be that something will fail and this feature will become a "blocker" to update to the new version

Reply:
> it can be that something will fail and this feature will become a "blocker" to update to the new version

Can you provide an example? We perform an action only when we encounter `XComArg` so `self.set_upstream` should work. If it doesn't work then it means that our implementation is corrupted.